File: q4/2.filterStarWarsMovies.py
import pandas as pd
import os

# Print the directory path
print("Current directory:", os.getcwd())
print("Change to q4 directory")
os.chdir("q4")
print("Current directory:", os.getcwd())


# Load the CSV 
df = pd.read_csv(".\extracted\TMDB_movie_dataset_v11.csv")

# Filter for Star Wars movies (case insensitive), note that this only filters the title column which contains 'Star Wars' in the title
star_wars_df = df[df["title"].str.contains("Star Wars", case=False, na=False)]

# Save filtered data
star_wars_df.to_csv("star_wars_movies.csv", index=False)

print("Filtered Star Wars movies saved as 'star_wars_movies.csv'")

File: q4/2.filterStarWarsMovies.js
import fs from 'fs';
import path from 'path';
import csv from 'csv-parser';


const inputFilePath = path.join('q4', 'extracted', 'TMDB_movie_dataset_v11.csv');
const outputFilePath = path.join('q4', 'star_wars_movies.csv');

console.log("Reading CSV file and filtering Star Wars movies...");

const starWarsMovies = [];

fs.createReadStream(inputFilePath)
  .pipe(csv())
  .on('data', (row) => {
    if (row.title && row.title.toLowerCase().includes('star wars')) {
      starWarsMovies.push(row);
    }
  })
  .on('end', () => {
    console.log("Saving filtered Star Wars movies...");
    const headers = Object.keys(starWarsMovies[0]).join(',') + '\n';
    const rows = starWarsMovies.map(movie => Object.values(movie).join(','));
    fs.writeFileSync(outputFilePath, headers + rows.join('\n'));
    console.log(`Filtered Star Wars movies saved as '${outputFilePath}'`);
  });
